index.js

```diff
@@ -33,7 +33,7 @@ exports.parse = function (str) {
 		// http://w3.org/TR/2012/WD-url-20120524/#collect-url-parameters
 		val = val === undefined ? null : decodeURIComponent(val);
 
-		if (ret[key] !== undefined) {
+		if (ret[key] === undefined) {
```

Comment:
See the commit message for the commit fixing the tests, but turns out refactoring to bind the tsame function broke ava's assertions somehow. Don't completely understand - stepped through and saw it hitting the `throw new assert.AssertionError ` line in ava/lib/assert.js, but that doesn't do anything - my javascript fu isn't strong enough.